### PR TITLE
Initial implementation of amp-inline-gallery

### PR DESCRIPTION
**Pull request by sparhami**
- Add amp-inline-gallery-pagination, for showing pagination dots/counts.
- Add amp-inline-gallery, to coordinate between amp-base-carousel and amp-inline-gallery-pagination
  * This will be expanded to handle captions and thumbnails in future PRs

To see relatively up to date demos, see: https://amp-testing-715cb.firebaseapp.com/manual/amp-inline-gallery/

### CODE REVIEW

**Comment by wassgha** [comment]
Any way we could split this into base carousel changes and inline gallery?

**Comment by sparhami** [comment]
> Any way we could split this into base carousel changes and inline gallery?

Yep, removed the amp-base-carousel changes and moved them to #25683.

File: test/manual/amp-inline-gallery/fixed-height.amp.html [hunk 1]
```diff
@@ -0,0 +1,100 @@
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <title>A Basic inline gallery</title>
+  <link rel="canonical" href="amps.html" >
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <link href='https://fonts.googleapis.com/css?family=Questrial' rel='stylesheet' type='text/css'>
```
Comment (wassgha) [inline test/manual/amp-inline-gallery/fixed-height.amp.html]: Is this needed? (in other files too)

**Comment by sparhami** [comment]
@jridgewell Added you as a reviewer for math.js (and anything else you want to review).

**Comment by sparhami** [comment]
Moved the `exponentialFalloff` refactor out of this PR, will send a separate one to keep this a bit cleaner.

**Pull request merged**